packages/cloud_firestore/cloud_firestore_web/lib/src/utils/exception.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright 2020, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:firebase_core/firebase_core.dart';
+
+import 'package:firebase/firebase.dart' as firebase;
+
+/// Returns a [FirebaseException] from a thrown web error.
+FirebaseException throwFirebaseException(Object exception) {
+  if (exception is! firebase.FirebaseError) {
+    return FirebaseException(
+        plugin: "cloud_firestore",
+        code: 'unknown',
+        message: 'An unknown error occurred.');
+  }
```

Comment:
If the original thrown exception is not a `firebase.FirebaseError` then the entire exception is lost here - so you would know what the original error was?